pyccel/parser/semantic.py

```diff
@@ -1932,6 +1932,8 @@ def _get_indexed_type(self, base, args, expr):
                     class_type = HomogeneousTupleType
                 elif dtype_cls is PythonList:
                     class_type = HomogeneousListType
+                elif dtype_cls is PythonSetFunction:
+                    class_type = HomogeneousSetType
```

Comment:
What do you think about using a dictionary here? I guess it should be more compact and faster than a series of if conditions

Reply:
That sounds like a good idea. I will implement it.

Reply:
Fixed : 7fa1e49